### PR TITLE
20199: Added enable_jumbo_frame to gateway, transit gateway and spoke gateway

### PR DESCRIPTION
**Pull request by mlin-aviatrix**

### CODE REVIEW

File: aviatrix/resource_aviatrix_gateway.go [hunk 1]
```diff
@@ -450,6 +450,12 @@ func resourceAviatrixGateway() *schema.Resource {
 				Optional:    true,
 				Description: "Whether to enforce Guard Duty IP blocking. Required when `enable_public_subnet_filtering` attribute is true. Valid values: true or false. Default value: true.",
 			},
+			"enable_jumbo_frame": {
+				Type:        schema.TypeBool,
+				Default:     true,
+				Optional:    true,
+				Description: "Enable jumbo frame support for Gateway. Valid values: true of false. Default value: true.",
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_gateway.go]: Typo: true of false -> true or false

File: aviatrix/resource_aviatrix_gateway.go [hunk 2]
```diff
@@ -968,6 +974,18 @@ func resourceAviatrixGatewayCreate(d *schema.ResourceData, meta interface{}) err
 		}
 	}
 
+	if d.Get("enable_jumbo_frame").(bool) {
+		err := client.EnableJumboFrame(gateway)
+		if err != nil {
+			return fmt.Errorf("couldn't enable jumbo frames for Gateway: %s", err)
+		}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_gateway.go]: Why call EnableJumboFrame if it is already enabled by default?

File: aviatrix/resource_aviatrix_spoke_gateway.go [hunk 3]
```diff
@@ -278,6 +278,12 @@ func resourceAviatrixSpokeGateway() *schema.Resource {
 				Optional:    true,
 				Description: "OOB subnet availability zone.",
 			},
+			"enable_jumbo_frame": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Enable jumbo frame support for transit gateway. Valid values: true of false. Default value: true.",
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_spoke_gateway.go]: Typo: true of false -> true or false

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 4]
```diff
@@ -399,6 +399,12 @@ func resourceAviatrixTransitGateway() *schema.Resource {
 				Optional:    true,
 				Description: "OOB subnet availability zone.",
 			},
+			"enable_jumbo_frame": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Enable jumbo frame support for spoke gateway. Valid values: true of false. Default value: true.",
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_transit_gateway.go]: Typo: true of false -> true or false

File: docs/resources/aviatrix_gateway.md [hunk 5]
```diff
@@ -263,6 +263,7 @@ The following arguments are supported:
 * `tag_list` - (Optional) Tag list of the gateway instance. Only available for AWS, AWSGov and Azure gateways. Example: ["key1:value1", "key2:value2"].
 * `enable_vpc_dns_server` - (Optional) Enable VPC DNS Server for gateway. Currently only supported for AWS and AWSGov gateways. Valid values: true, false. Default value: false.
 * `zone` - (Optional) Availability Zone. Only available for Azure and Public Subnet Filtering gateway. Available for Azure as of provider version R2.17+.
+* `enable_jumbo_frame` - (Optional) Enable jumbo frames for the gateway. Default value is true.
```
Comment (CyrusJavan) [inline docs/resources/aviatrix_gateway.md]: Add documentation for spoke and transit as well.

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 6]
```diff
@@ -2291,6 +2314,20 @@ func resourceAviatrixTransitGatewayUpdate(d *schema.ResourceData, meta interface
 		}
 	}
 
+	if d.HasChange("enable_jumbo_frame") {
+		if d.Get("enable_jumbo_frame").(bool) {
+			err := client.EnableJumboFrame(gateway)
+			if err != nil {
+				return fmt.Errorf("could not enable jumbo frame for gateway: %v", err)
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_transit_gateway.go]: Error messages need to be unique strings for debugging/diagnosing issues. You have used the same string `could not enable jumbo frame for gateway` across the 3 gateway resources, if a customer reported a problem and all we had was this error string, we would have no clue which gateway resource had the problem. Also you reused the same string for Create and Update, which is also ambiguous.

Add more context to make the strings unique, for example, this is Transit Gateway update so the error should be
```
could not enable jumbo frame for transit gateway when updating
```

**Review by CyrusJavan — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_transit_gateway.go (hunk 6 above)
Comment (mlin-aviatrix) [reply aviatrix/resource_aviatrix_transit_gateway.go]: I added unique messages for the error messages so you can tell if it's a gateway, spoke gateway or transit gateway. Also differentiated create and update error messages.

**Review by CyrusJavan — APPROVED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**